Task8/README.md

```diff
@@ -0,0 +1,158 @@
+And since to download data from `BigQuery` I used the recommended `pandas_gbq` (not included in zenML integrations) it fails when submitting the job to 
+`Vertex AI` (because zenML defined an image where they install only their integrated packages).
+
+Correction:
+
+That is true by default, but when you are registering a new step-operator you can define your custom base image `--base_image=<CUSTOM_BASE_IMAGE>` 
+for creating an environment to run your jobs on Vertex AI. (so you can download any packages you need, not only the ones included by default).
+
+## ZenML pipeline downloading data from BigQuery
+
+Please view `pipeline.py` for a description on how to do it.
+
+## Running ZenML pipeline in Vertex AI pipelines
+
+### 1. Configure your GCP project
+
+1.1. `gcloud cli` set up
+
+Make sure you are in the correct project `gcloud config list`
+
+1.2. Make sure you have the right permissions to create and manage Vertex AI custom jobs.
+
+Create a service account
+```
+gcloud iam service-accounts create NAME
+
+#EXAMPLE
+gcloud iam service-accounts create zenml-sa
+```
+Grant roles to the service account
+```
+gcloud projects add-iam-policy-binding <PROJECT_ID> --member="serviceAccount:<SA-NAME>@<PROJECT_ID>.iam.gserviceaccount.com" --role=<ROLE>
+
+#EXAMPLE
+gcloud projects add-iam-policy-binding zenml-vertex-ai --member="serviceAccount:zenml-sa@zenml-vertex-ai.iam.gserviceaccount.com" --role=roles/storage.admin
+gcloud projects add-iam-policy-binding zenml-vertex-ai --member="serviceAccount:zenml-sa@zenml-vertex-ai.iam.gserviceaccount.com" --role=roles/aiplatform.admin
+```
+Generate the key file (it generates in the same location)
+```
+gcloud iam service-accounts keys create <FILE-NAME>.json --iam-account=<SA-NAME>@<PROJECT_ID>.iam.gserviceaccount.com
+
+EXAMPLE
+gcloud iam service-accounts keys create zenml-sa-file.json --iam-account=zenml-sa@zenml-vertex-ai.iam.gserviceaccount.com
+```
+Set the environment variable:
+
+To use service accounts with the Google Cloud CLI, you need to set an environment variable where your code runs
+```
+export GOOGLE_APPLICATION_CREDENTIALS=<KEY-FILE-LOCATION>
+
+#EXAMPLE
+export GOOGLE_APPLICATION_CREDENTIALS="/Users/co2yd4bcjgh6/Library/CloudStorage/OneDrive-Personal/myWorkspace/github/DataScience/Task8/zenml-sa-file.json"
+```
+1.3. Create a `GCP bucket` to which Vertex should output any artifacts from the training run
+```
+gsutil mb -l <REGION> gs://bucket-name
+
+#EXAMPLE
+gsutil mb -l europe-west1 gs://zenml-bucket
+```
+
+1.4. Configure and enable container registry in GCP
+
+This registry will be used by ZenML to push your job images that Vertex will use.
+
+a) [Enable](https://cloud.google.com/container-registry/docs) Container Registry
+b) [Authenticate](https://cloud.google.com/container-registry/docs/advanced-authentication) your local `docker` cli with your GCP container registry 
+
+```
+docker pull busybox
```

Comment:
is this image used or just an example?

Reply:
Yes, it is. As I understand it, this image is not used in any way. You push the image, which creates a repository in the Container registry, and `zenml` that uses this repository to push their job images that are then used by `Vertex`.